sklearn/metrics/cluster/tests/test_unsupervised.py

```diff
@@ -341,14 +344,9 @@ def test_davies_bouldin_score():
     pytest.approx(davies_bouldin_score(X, labels), 2 * np.sqrt(0.5) / 3)
 
     # Ensure divide by zero warning is not raised in general case
-    with pytest.warns(None) as record:
+    with warnings.catch_warnings():
+        warnings.simplefilter("error", UndefinedMetricWarning)
```

Comment:
According to the ``davies_bouldin_score`` code, there's a special case to avoid dividing a numpy array by 0. So we're expecting a RuntimeWarning.
```suggestion
        warnings.simplefilter("error", RuntimeWarning)
```